compiler/src/dotty/tools/dotc/Run.scala

```diff
@@ -400,7 +460,101 @@ class Run(comp: Compiler, ictx: Context) extends ImplicitRunInfo with Constraint
 }
 
 object Run {
+
+  class SubPhases(val phase: Phase):
```

Comment:
The reason this exists is because `TyperPhase` is really like three phases, it might be good to allow cancellation in the middle of the phase

Reply:
is there any reason not to split `TyperPhase` into three phases?